core/include/scipp/core/variable.h

```diff
@@ -704,18 +727,34 @@ class SCIPP_CORE_EXPORT VariableProxy : public VariableConstProxy {
   // returning `a += b` but I am not sure how Pybind11 handles object lifetimes
   // (would this suffer from the same issue?).
   template <class T> VariableProxy assign(const T &other) const;
+
   VariableProxy operator+=(const Variable &other) const;
   VariableProxy operator+=(const VariableConstProxy &other) const;
-  VariableProxy operator+=(const double value) const;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  VariableProxy operator+=(const T v) const {
+    return *this += makeVariable<T>(v);
+  }
+
   VariableProxy operator-=(const Variable &other) const;
   VariableProxy operator-=(const VariableConstProxy &other) const;
-  VariableProxy operator-=(const double value) const;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  VariableProxy operator-=(const T v) const {
+    return *this -= makeVariable<underlying_type_t<T>>(v);
+  }
+
   VariableProxy operator*=(const Variable &other) const;
   VariableProxy operator*=(const VariableConstProxy &other) const;
-  VariableProxy operator*=(const double value) const;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  VariableProxy operator*=(const T v) const {
+    return *this *= makeVariable<underlying_type_t<T>>(v);
+  }
+
   VariableProxy operator/=(const Variable &other) const;
   VariableProxy operator/=(const VariableConstProxy &other) const;
-  VariableProxy operator/=(const double value) const;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  VariableProxy operator/=(const T v) const {
+    return *this /= makeVariable<underlying_type_t<T>>(v);
+  }
```

Comment:
One day someone should sit down and find a way to avoid this code duplication between container and proxies. Maybe a mixin class can help? (this is not a reuqest to do it here, just something to think about when taking a shower)

Reply:
I opened https://github.com/scipp/scipp/issues/586 to keep track of this (and any more cases that are found).